pkg/builder/action_merge_meta_data_builder.go

```diff
@@ -190,47 +200,19 @@ func (rb *ResizeMergePolicyBuilder) Build() (*proto.ActionMergePolicyDTO, error)
 
 // Horizontal Scale Merge Policy DTO builder
 type HorizontalScaleMergePolicyBuilder struct {
-	entityType                *proto.EntityDTO_EntityType
-	chainedAggregationTargets []*ActionDeDuplicateAndAggregationTargetBuilder
-	commTypes                 []*CommodityMergeData
-	entityFilters             []*proto.EntityDTO_ContainerPodData
-}
-
-func NewHorizontalScaleMergeSpecBuilder() *HorizontalScaleMergePolicyBuilder {
-	return &HorizontalScaleMergePolicyBuilder{}
-}
-
-func (hsb *HorizontalScaleMergePolicyBuilder) ForEntityType(entityType proto.EntityDTO_EntityType) *HorizontalScaleMergePolicyBuilder {
-	hsb.entityType = &entityType
-	return hsb
+	*MergePolicyBuilder
+	entityFilters []*proto.EntityDTO_ContainerPodData
 }
 
-func (hsb *HorizontalScaleMergePolicyBuilder) DeDuplicateAndAggregateBy(mergeTarget *ActionDeDuplicateAndAggregationTargetBuilder) *HorizontalScaleMergePolicyBuilder {
-	hsb.chainedAggregationTargets = append(hsb.chainedAggregationTargets, mergeTarget)
-	return hsb
-}
-
-func (hsb *HorizontalScaleMergePolicyBuilder) ForContainerPodDataFilter(podData proto.EntityDTO_ContainerPodData) *HorizontalScaleMergePolicyBuilder {
-	hsb.entityFilters = append(hsb.entityFilters, &podData)
-	return hsb
-}
-
-func (hsb *HorizontalScaleMergePolicyBuilder) ForCommodity(commType proto.CommodityDTO_CommodityType) *HorizontalScaleMergePolicyBuilder {
-	comm := &CommodityMergeData{
-		commType: commType,
+func NewHorizontalScaleMergePolicyBuilder() *HorizontalScaleMergePolicyBuilder {
+	return &HorizontalScaleMergePolicyBuilder{
+		MergePolicyBuilder: NewMergePolicyBuilder(),
 	}
-	hsb.commTypes = append(hsb.commTypes, comm)
-	return hsb
 }
 
-func (hsb *HorizontalScaleMergePolicyBuilder) ForCommodityAndAttribute(commType proto.CommodityDTO_CommodityType,
-	changedAttr proto.ActionItemDTO_CommodityAttribute) *HorizontalScaleMergePolicyBuilder {
-	comm := &CommodityMergeData{
-		commType:    commType,
-		changedAttr: changedAttr,
-	}
-	hsb.commTypes = append(hsb.commTypes, comm)
-	return hsb
+func (hsmb *HorizontalScaleMergePolicyBuilder) ForContainerPodDataFilter(podData proto.EntityDTO_ContainerPodData) *HorizontalScaleMergePolicyBuilder {
```

Comment:
This should also move to `MergePolicyBuilder`.

Reply:
Thanks!  refactored the code.